packages/compiler/core/checker.ts

```diff
@@ -935,44 +955,24 @@ export function createChecker(program: Program): Checker {
         if (symbolLinks.declaredType) {
           baseType = symbolLinks.declaredType;
         } else {
-          baseType =
```

Comment:
For review: I just refactored this into its own function as we had that reused exactly the same right below